recipes/Media/ASR/seq2seq/train.py

```diff
@@ -0,0 +1,346 @@
+#!/usr/bin/env python3
+
+# ################################
+# Authors modifier: Gaelle Laperriere
+# ################################
+
+import sys
+import torch
+import logging
+import speechbrain as sb
+from hyperpyyaml import load_hyperpyyaml
+from speechbrain.dataio.batch import PaddedBatch
+
+"""
+Recipe for training a sequence-to-sequence ASR system with Media.
+
+To run this recipe, do the following:
+> python train.py hparams/train.yaml
+
+With the default hyperparameters, the system employs a VanillaNN decoder.
+
+The neural network is trained on greedy CTC.
+
+The experiment file is flexible enough to support a large variety of
+different systems. By properly changing the parameter files, you can try
+different encoders, decoders,
+training tasks (Media , PortMedia), and many other possible variations.
+"""
+
+logger = logging.getLogger(__name__)
+
+
+# Define training procedure.
+class ASR(sb.core.Brain):
+    def compute_forward(self, wavs, wav_lens, stage):
+        """Forward computations from waveform to output probabilities."""
+
+        # Forward pass.
+        feats = self.hparams.compute_features(wavs)
+        feats = self.modules.normalize(feats, wav_lens)
+        x_enc = self.modules.dnn(feats.detach())
+        x = self.modules.dec(x_enc)
+
+        # Output layer for seq2seq log-probabilities.
+        logits = self.modules.output_lin(x)
+        p_ctc = self.hparams.softmax(logits)
+
+        return p_ctc, wav_lens
+
+    def compute_objectives(self, predictions, ids, chars, char_lens, stage):
+        """Computes the loss (CTC+NLL) given predictions and targets."""
+
+        # Get predictions & loss.
+        p_ctc, wav_lens = predictions
+        loss = self.hparams.ctc_cost(p_ctc, chars, wav_lens, char_lens)
+
+        # Get metrics.
+        if stage != sb.Stage.TRAIN:
+            # Generate sequences with CTC greedy decoder.
+            sequence = sb.decoders.ctc_greedy_decode(
+                p_ctc, wav_lens, self.hparams.blank_index
+            )
+            # Update metrics.
+            self.cer_metric.append(
+                ids=ids,
+                predict=sequence,
+                target=chars,
+                target_len=char_lens,
+                ind2lab=self.label_encoder.decode_ndim,
+            )
+            self.ctc_metric.append(ids, p_ctc, chars, wav_lens, char_lens)
+
+        return loss
+
+    def init_optimizers(self):
+        """Initializes the model optimizer"""
+
+        # Join optimizers.
+        self.optimizer = self.hparams.opt_class(self.hparams.model.parameters())
+
+        # Add opitmizers to checkpoint recoverables.
+        if self.checkpointer is not None:
+            self.checkpointer.add_recoverable("optimizer", self.optimizer)
+
+    def fit_batch(self, batch):
```

Comment:
Why do you need to overload fit_batch and evaluate_batch ?

Reply:
Same as above, always did... Because of the recipe I took example from. Do I need to remove it?